src/test/java/org/prebid/server/bidder/stroeercore/StroeerCoreBidderTest.java

```diff
@@ -0,0 +1,401 @@
+        assertThat(result.getValue()).containsOnly(BidderBid.of(expectedBid1, BidType.banner, "EUR"),
+                BidderBid.of(expectedBid2, BidType.banner, "EUR"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnErrorIfResponseBodyCouldNotBeParsed() {
+        // given
+        final HttpCall<BidRequest> httpCall = createHttpCall("[]");
+
+        // when
+        final Result<List<BidderBid>> result = bidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors())
+                .hasSize(1)
+                .first()
+                .satisfies(error -> assertThat(error.getType()).isEqualTo(BidderError.Type.bad_server_response))
+                .satisfies(error -> assertThat(error.getMessage()).startsWith("Failed to decode"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyListIfZeroBids() throws JsonProcessingException {
+        // given
+        final HttpCall<BidRequest> httpCall = createHttpCall(StroeerCoreBidResponse.of(emptyList()));
+
+        // when
+        final Result<List<BidderBid>> result = bidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    private BidRequest createBidRequest(final String... slotIds) {
+        final ArrayList<Imp> imps = new ArrayList<>();
+
+        long impId = 1;
+        for (String slotId : slotIds) {
+            imps.add(createImp(String.valueOf(impId++), slotId));
+        }
+
+        return createBidRequest(imps.toArray(new Imp[]{}));
+    }
+
+    private BidRequest createBidRequest(final Imp... imps) {
+        return BidRequest.builder().imp(List.of(imps)).build();
+    }
+
+    private Imp createImp(final String id, final String slotId) {
+        final ObjectNode impExtNode = mapper.valueToTree(ExtPrebid.of(null, ExtImpStroeerCore.of(slotId)));
+        ImpBuilder impBuilder = Imp.builder();
+        impBuilder.id(String.valueOf(id));
+        impBuilder.banner(Banner.builder().build());
+        impBuilder.ext(impExtNode);
+        return impBuilder.build();
+    }
+
+    private Imp createNonParsableImpFrom(final Imp validImp) {
+        return validImp.toBuilder()
+                .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode()))).build();
+    }
+
+    private Imp createNullBannerImpFrom(final Imp validImp) {
+        return validImp.toBuilder()
+                .banner(null).video(Video.builder().build()).build();
+    }
+
+    private Imp createEmptySlotIdImpFrom(final Imp validImp) {
+        return validImp.toBuilder()
+                .ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpStroeerCore.of(" ")))).build();
+    }
+
+    private Imp createImpWithFloorFrom(final Imp imp, final String bidFloorCurrency, final BigDecimal bidFloor) {
+        return imp.toBuilder().bidfloorcur(bidFloorCurrency).bidfloor(bidFloor).build();
+    }
```

Comment:
I think these are superfluous methods and they only complicate the tests.

What about this approach?
```
    private Imp createImp(UnaryOperator<Imp.ImpBuilder> impCustomizer) {
        return impCustomizer
                .apply(Imp.builder().banner(Banner.builder().build()))
                .build();
    }

    private Imp createImp(String slotId, UnaryOperator<Imp.ImpBuilder> impCustomizer) {
        final ObjectNode impExtNode = mapper.valueToTree(ExtPrebid.of(null, ExtImpStroeerCore.of(slotId)));

        final UnaryOperator<Imp.ImpBuilder> addImpExt = imp -> imp.ext(impExtNode);
        return createImp(addImpExt.andThen(impCustomizer)::apply);
    }
```

Than we can easily rewrite (as example) `makeHttpRequestsShouldIgnoreInvalidImpressions` `given` part:
```
        final BidRequest bidRequest = createBidRequest(
                createImp(imp -> imp.ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode())))),
                createImp("b", identity()),
                createImp("   ", identity()),
                createImp(imp -> imp.banner(null)),
                createImp("e", imp -> imp.bidfloor(BigDecimal.ONE).bidfloorcur("GPB")),
                createImp("f", identity()));
```